memory/node.go

```diff
@@ -187,3 +188,91 @@ func LinkCast(v interface{}) (l Link, ok bool) {
 	}
 	return l, true
 }
+
+func (n Node) Read(fun stream.ReadFun) error {
+	err := read(n, fun, []interface{}{})
+	if err == stream.NodeReadAbort {
+		err = nil
+	}
+	return err
+}
+
+func read(curr interface{}, fun stream.ReadFun, path []interface{}) error {
```

Comment:
Not much can be changed, it's just going through the datastructure. I could improve the NodeReadSkip lines though